packages/docregistry/src/context.ts

```diff
@@ -96,6 +88,42 @@ export class Context<T extends DocumentRegistry.IModel>
     return this._fileChanged;
   }
 
+  /**
+   * A signal emitted when the document state changes.
+   */
+  get stateChanged(): ISignal<this, IChangedArgs<any>> {
```

Comment:
In an effort to simplify the document model, I've moved `stateChanged`, `dirty`, and `readOnly` to the document context. The rationale is that these are information about how the document relates to the server, not about its actual content. We may need to revisit this when we think about multi-user saving, but the conversion was pretty smooth.

Reply:
I would say there are multiple distinct points to "read only":
1. Can the user write (the current file name) to disk?
2. Can the user write to the datastore?
3. Does the UI allow editing (code editor read only).

In my mind, point 3 should depend directly on point 2.

Reply:
True, in the past `readOnly` has *only* referred to point 1, and points 2 (with a modeldb) and 3 have been ignored. Perhaps we should revisit.